src/rosdistro/manifest_provider/github.py

```diff
@@ -39,32 +39,24 @@
     from urllib2 import URLError
 
 from rosdistro import logger
-from rosdistro.manifest_provider import get_release_tag
-from rosdistro.manifest_provider.git import check_remote_tag_exists
 
 
 def github_manifest_provider(_dist_name, repo, pkg_name):
     assert repo.version
-    if 'github.com' not in repo.url:
+    server, path = repo.get_url_parts()
+    if server != 'github.com':
         logger.debug('Skip non-github url "%s"' % repo.url)
         raise RuntimeError('can not handle non github urls')
 
-    release_tag = get_release_tag(repo, pkg_name)
+    release_tag = repo.get_release_tag(pkg_name)
 
-    if not check_remote_tag_exists(repo.url, release_tag):
+    if release_tag not in repo.remote_tags:
```

Comment:
I suppose I haven't really made good on it, but the thought was that there would be other places where it would be valuable to be able to iterate on the remote tags, for example:

https://github.com/ros-infrastructure/rosdistro/blob/856aa03ce8b9517c18e5a0787d8f5f2420ecf7eb/src/rosdistro/freeze_source.py#L90

FWIW, this API would allow testing membership without getting the full list, albeit with a little more magic behind the scenes: https://docs.python.org/2/reference/datamodel.html#object.__contains__